### DISCUSSION TITLE
ValueError: Model name MyModel was not found in search namespaces: ['InnerEye.ML.configs'].

### DISCUSSION THREAD

**Discussion by Med-Rokaimi** [microsoft/InnerEye-DeepLearning#817] [Q&A] [answered]
Hey,
I'm trying to run  Glaucoma detection model following the steps in the link
[Sample task](https://innereye-deeplearning.readthedocs.io/md/sample_tasks.html).
I set up the InnerEye-DeepLearningnner locally.
I created MyGlaucomaModel in the path (InnerEye/ML/configs), and my dataset folder is located in the same path as well.
This is the model I created:
<pre>class MyGlaucomaModel(GlaucomaPublic):
    def __init__(self) -> None:
        super().__init__()
        local_dataset=full_ml_test_data_path(),
        self.azure_dataset_id="glaucoma_detection",</pre>

When I try to run the model by this command: 
<pre>python InnerEye/ML/runner.py --model=MyGlaucomaModel</pre>
I got this error:

<pre>Starting InnerEye runner at InnerEye/ML/runner.py
Adding InnerEye folder to sys.path: /home/ai/InnerEye-DeepLearning
Setting up logging to stdout.
Setting logging level to 20 (INFO)
2022-10-25T16:48:21Z INFO     rpdb is handling traps. To debug: identify the main runner.py process, then as root: kill -TRAP &lt;process_id&gt;; nc 127.0.0.1 4444
2022-10-25T16:48:21Z WARNING  (from attempt to import module InnerEye.ML.configs.MyGlaucoma): name &apos;full_ml_test_data_path&apos; is not defined
2022-10-25T16:48:21Z WARNING  (from attempt to import module InnerEye.ML.configs.classification.GlaucomaPublic): cannot import name &apos;GlaucomaPublic&apos; from partially initialized module &apos;InnerEye.ML.configs.classification.GlaucomaPublic&apos; (most likely due to a circular import) (/home/ai/InnerEye-DeepLearning/InnerEye/ML/configs/classification/GlaucomaPublic.py)
Traceback (most recent call last):
  File &quot;InnerEye/ML/runner.py&quot;, line 467, in &lt;module&gt;
    main()
  File &quot;InnerEye/ML/runner.py&quot;, line 461, in main
    run(project_root=fixed_paths.repository_root_directory(),
  File &quot;InnerEye/ML/runner.py&quot;, line 457, in run
    return runner.run()
  File &quot;InnerEye/ML/runner.py&quot;, line 215, in run
    self.parse_and_load_model()
  File &quot;InnerEye/ML/runner.py&quot;, line 164, in parse_and_load_model
    config_or_container = model_config_loader.create_model_config_from_name(model_name=azure_config.model)
  File &quot;/home/ai/InnerEye-DeepLearning/InnerEye/ML/utils/config_loader.py&quot;, line 132, in create_model_config_from_name
    raise ValueError(
ValueError: Model name MyGlaucomaModel was not found in search namespaces: [&apos;InnerEye.ML.configs&apos;].</pre>
Can you please help me? I

**Comment by ant0nsc** [comment, marked as answer]
@Med-Rokaimi, the problem is that the Python code can't be imported. Look at the two warnings:
```
from attempt to import module InnerEye.ML.configs.MyGlaucoma): name 'full_ml_test_data_path' is not defined
from attempt to import module InnerEye.ML.configs.classification.GlaucomaPublic): cannot import name 'GlaucomaPublic' from partially initialized module 'InnerEye.ML.configs.classification.GlaucomaPublic' (most likely due to a circular import)
```
You need to fix those 2 errors before you will be able to proceed.
